From 59545089e8b9e1d1dfc6d05b040f02027b681137 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Thu, 4 Jul 2019 22:12:05 -0700
Subject: [PATCH 0636/1852] v4l: xilinx: sdirxss: Add support to enumerate
 media bus formats

commit 302f4bd671cf9a1ee9f78055190ed81493df5b17 from
https://github.com/Xilinx/linux-xlnx.git

Add support for enum_mbus_code v4l subdev call to enumerate the media
bus formats supported. These are YUV 422 and YUV 420 10 bpc.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../media/platform/xilinx/xilinx-sdirxss.c    | 26 +++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/drivers/media/platform/xilinx/xilinx-sdirxss.c b/drivers/media/platform/xilinx/xilinx-sdirxss.c
index b1259e52cdfe..4e060ad796fc 100644
--- a/drivers/media/platform/xilinx/xilinx-sdirxss.c
+++ b/drivers/media/platform/xilinx/xilinx-sdirxss.c
@@ -322,6 +322,11 @@ static const char * const xsdirxss_clks[] = {
 	"s_axi_aclk", "sdi_rx_clk", "video_out_clk",
 };
 
+static const u32 xsdirxss_mbus_fmts[] = {
+	MEDIA_BUS_FMT_UYVY10_1X20,
+	MEDIA_BUS_FMT_VYYUYY10_4X20,
+};
+
 static inline struct xsdirxss_state *
 to_xsdirxssstate(struct v4l2_subdev *subdev)
 {
@@ -1346,6 +1351,26 @@ static int xsdirxss_set_format(struct v4l2_subdev *sd,
 	return 0;
 }
 
+/**
+ * xsdirxss_enum_mbus_code - Handle pixel format enumeration
+ * @sd: pointer to v4l2 subdev structure
+ * @cfg: V4L2 subdev pad configuration
+ * @code: pointer to v4l2_subdev_mbus_code_enum structure
+ *
+ * Return: -EINVAL or zero on success
+ */
+static int xsdirxss_enum_mbus_code(struct v4l2_subdev *sd,
+				   struct v4l2_subdev_pad_config *cfg,
+				   struct v4l2_subdev_mbus_code_enum *code)
+{
+	if (code->pad || code->index >= ARRAY_SIZE(xsdirxss_mbus_fmts))
+		return -EINVAL;
+
+	code->code = xsdirxss_mbus_fmts[code->index];
+
+	return 0;
+}
+
 /**
  * xsdirxss_open - Called on v4l2_open()
  * @sd: Pointer to V4L2 sub device structure
@@ -1520,6 +1545,7 @@ static const struct v4l2_subdev_video_ops xsdirxss_video_ops = {
 static const struct v4l2_subdev_pad_ops xsdirxss_pad_ops = {
 	.get_fmt = xsdirxss_get_format,
 	.set_fmt = xsdirxss_set_format,
+	.enum_mbus_code = xsdirxss_enum_mbus_code,
 };
 
 static const struct v4l2_subdev_ops xsdirxss_ops = {
-- 
2.31.1

